src/services/__tests__/scheduler-service.test.ts

```diff
@@ -0,0 +1,64 @@
+import 'reflect-metadata'
+import { SchedulerService } from '../scheduler-service'
+import { config } from 'node-config-ts'
+import { container } from 'tsyringe'
+import { jest } from '@jest/globals'
+
+describe('scheduler service', () => {
+  jest.setTimeout(20000)
+  beforeAll(async () => {
+    container.registerInstance('config', Object.assign({}, config, { schedulerIntervalMS: 3000 }))
+    container.registerSingleton('schedulerService', SchedulerService)
+  })
+
+  test('will run the task repeatedly', (done) => {
+    const numberOfRunsBeforeDone = 3
+    const schedulerService = container.resolve<SchedulerService>('schedulerService')
+
+    let count = 0
+    const task = jest.fn(() => {
+      if (count === numberOfRunsBeforeDone) {
+        schedulerService.stop()
+        runChecks()
+      }
+
+      count = count + 1
+      return Promise.resolve()
+    })
+
+    schedulerService.start(task)
+    const runChecks = () => {
```

Comment:
is it safe to define this function after calling `scheduleService.start` with a task that may try to call it?

Reply:
given the single-thread nature of JS it probably actually is perfectly safe, but for my C++ trained brain that is paranoid about thread safety, could you declare the `runChecks` function before calling `schedulerService.start()`?  Would make the test flow more intuitive as well IMO

Reply:
It does feel really weird to use something before its defined. I have changed the tests so it no longer does this haha.